controllers/upgrade.go

```diff
@@ -399,7 +399,8 @@ func (r *PravegaClusterReconciler) syncSegmentStoreVersion(p *pravegav1beta1.Pra
 	}
 
 	if ready {
-		pod, err := r.getOneOutdatedPod(sts, p.Status.TargetVersion)
+		labels := p.LabelsForPravegaCluster()
```

Comment:
Would this function fetch new or old labels?

Reply:
`LabelsForPravegaCluster()` is a subset of labels and that will be present  as mandatory labels in every sts.